### ISSUE TITLE
[llvm-amdgpu] ...what? Backslashes?

### ISSUE THREAD

**Issue by tildearrow** [rocm-arch/rocm-arch#156]
I cannot compile llvm-amdgpu because there is a Windows backslash somewhere:

```
$ makepkg -sri
==> Making package: llvm-amdgpu 3.3.0-2 (Fri 08 May 2020 02:05:09 PM -05)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Found rocm-ocl-3.3.0.tar.gz
==> Validating source files with sha256sums...
    rocm-ocl-3.3.0.tar.gz ... Passed
==> Extracting sources...
  -> Extracting rocm-ocl-3.3.0.tar.gz with bsdtar
==> Removing existing $pkgdir/ directory...
==> Starting build()...
-- clang project is enabled
-- clang-tools-extra project is disabled
-- compiler-rt project is disabled
-- debuginfo-tests project is disabled
-- libc project is disabled
-- libclc project is disabled
-- libcxx project is disabled
-- libcxxabi project is disabled
-- libunwind project is disabled
-- lld project is enabled
-- lldb project is disabled
-- mlir project is disabled
-- openmp project is disabled
-- parallel-libs project is disabled
-- polly project is disabled
-- pstl project is disabled
-- Native target architecture is X86
-- Threads enabled.
-- Doxygen disabled.
-- Go bindings disabled.
-- Could NOT find OCaml (missing: OCAMLFIND OCAML_VERSION OCAML_STDLIB_PATH) 
-- OCaml bindings disabled.
-- LLVM host triple: x86_64-pc-linux-gnu
-- LLVM default target triple: x86_64-pc-linux-gnu
/usr/bin/ar: creating t.a
-- Building with -fPIC
-- Constructing LLVMBuild project information
CMake Error at cmake/modules/AddLLVM.cmake:498 (add_library):
  Syntax error in cmake code when parsing string

    /usr/include/GL\glew.h

  Invalid character escape '\g'.
Call Stack (most recent call first):
  cmake/modules/AddLLVM.cmake:709 (llvm_add_library)
  cmake/modules/AddLLVM.cmake:695 (add_llvm_library)
  lib/Support/CMakeLists.txt:54 (add_llvm_component_library)


-- Configuring incomplete, errors occurred!
See also "/rocm-arch/llvm-amdgpu/src/CMakeFiles/CMakeOutput.log".
See also "/rocm-arch/llvm-amdgpu/src/CMakeFiles/CMakeError.log".
==> ERROR: A failure occurred in build().
    Aborting...
```

I've been searching the code for any backslashes but I couldn't find any...

**Comment by acxz** [comment]
I kinda have a feeling that your `GL` directory is messed up. 
Can you show the output of `pacman -Qo /usr/include/GL*/*`
and `pacman -Qo /usr/include/GL*\\*`?

**Comment by tildearrow** [comment]
Sure.

`pacman -Qo /usr/include/GL*/*`:
```
/usr/include/GL/eglew.h is owned by glew 2.2.0-2
/usr/include/GLES2/gl2ext.h is owned by libglvnd 1.3.1-1
/usr/include/GLES2/gl2.h is owned by libglvnd 1.3.1-1
/usr/include/GLES2/gl2platform.h is owned by libglvnd 1.3.1-1
/usr/include/GLES3/gl31.h is owned by libglvnd 1.3.1-1
/usr/include/GLES3/gl32.h is owned by libglvnd 1.3.1-1
/usr/include/GLES3/gl3ext.h is owned by libglvnd 1.3.1-1
/usr/include/GLES3/gl3.h is owned by libglvnd 1.3.1-1
/usr/include/GLES3/gl3platform.h is owned by libglvnd 1.3.1-1
/usr/include/GLES/egl.h is owned by mesa 19.0.6-1
/usr/include/GLES/glext.h is owned by mesa 19.0.6-1
/usr/include/GLES/gl.h is owned by mesa 19.0.6-1
/usr/include/GLES/glplatform.h is owned by mesa 19.0.6-1
/usr/include/GL/freeglut_ext.h is owned by freeglut 3.2.1-1
/usr/include/GL/freeglut.h is owned by freeglut 3.2.1-1
/usr/include/GL/freeglut_std.h is owned by freeglut 3.2.1-1
/usr/include/GL/freeglut_ucall.h is owned by freeglut 3.2.1-1
/usr/include/GLFW/glfw3.h is owned by glfw-x11 3.3.2-1
/usr/include/GLFW/glfw3native.h is owned by glfw-x11 3.3.2-1
/usr/include/GL/glcorearb.h is owned by libglvnd 1.3.1-1
/usr/include/GL/glew.h is owned by glew 2.2.0-2
/usr/include/GL/glext.h is owned by libglvnd 1.3.1-1
/usr/include/GL/gl.h is owned by libglvnd 1.3.1-1
/usr/include/GL/gl_mangle.h is owned by mesa 19.0.6-1
/usr/include/GL/glu.h is owned by glu 9.0.1-1
/usr/include/GL/glu_mangle.h is owned by glu 9.0.1-1
/usr/include/GL/glut.h is owned by freeglut 3.2.1-1
/usr/include/GL/glxew.h is owned by glew 2.2.0-2
/usr/include/GL/glxext.h is owned by libglvnd 1.3.1-1
/usr/include/GL/glx.h is owned by libglvnd 1.3.1-1
/usr/include/GL/glxint.h is owned by xorgproto 2020.1-1
/usr/include/GL/glx_mangle.h is owned by mesa 19.0.6-1
/usr/include/GL/glxmd.h is owned by xorgproto 2020.1-1
/usr/include/GL/glxproto.h is owned by xorgproto 2020.1-1
/usr/include/GL/glxtokens.h is owned by xorgproto 2020.1-1
/usr/include/GL/internal/ is owned by mesa 19.0.6-1
/usr/include/GL/internal/ is owned by xorgproto 2020.1-1
/usr/include/GL/osmesa.h is owned by mesa 19.0.6-1
/usr/include/GL/wglew.h is owned by glew 2.2.0-2
```
`pacman -Qo /usr/include/GL*\\*`:
```
error: No package owns /usr/include/GL\glew.h
```

Apparently it looks fine.

**Comment by tildearrow** [comment]
While checking for this, I found a file that looked odd: `/usr/include/GLglew.h` and wasn't owned by any packages.
However, the problem persists after removing it.

Wait, I found another file called `/usr/include/GL\glew.h` (yes, with a backslash in its name). I remember adding this symlink out of rage 2 years ago when I was trying to compile a project that used backslashes in an include directive... After removing it, it seems to work now.

Sorry for opening this. It is user issue.
